Add translation support to RichTextLabel

Animtim opened the pull request:
add translation support to RichTextLabel, and respect its Auto Translate setting

Fix https://github.com/godotengine/godot/issues/34050

Animtim committed:
Add translation support to RichTextLabel

BUG: 34050
add translation support to RichTextLabel, and respect its Auto Translate setting

bruvzg commented on scene/gui/rich_text_label.cpp:
```diff
@@ -1793,6 +1794,17 @@ void RichTextLabel::_notification(int p_what) {
 		case NOTIFICATION_TRANSLATION_CHANGED: {
 			_stop_thread();
 			main->first_invalid_line.store(0); //invalidate ALL
```
These lines are no longer necessary, `clear` and `parse_bbcode` will do it automatically.

bruvzg reviewed (commented):
There are other functions that are resetting text, and should be updated in the same way as well: `set_effects`, `install_effect`, `_set`, `set_use_bbcode`.

Also, since `xl_text` is not used anywhere outside the function it is set, there's no reason to have it in a class, it can be a local variable.

Animtim commented:
Thanks @bruvzg for the comments. I've removed the no-longer-necessary lines, refactored a bit my previous changes to avoid repetitions, and updated `set_effects`, `install_effect` and `set_use_bbcode.` About `_set`, I don't think it actually needs to be updated as it only calls `set_text` if needed, which already takes care of the translation part.

KoBeWi commented on scene/gui/rich_text_label.cpp:
```diff
@@ -5103,13 +5103,22 @@ void RichTextLabel::set_text(const String &p_bbcode) {
+}
+
+void RichTextLabel::_apply_translated_bbcode(const String &p_bbcode) {
+	String xl_text;
+	if (is_auto_translating()) {
+		xl_text = tr(p_bbcode);
+	} else {
+		xl_text = p_bbcode;
+	}
```
```suggestion
	String xl_text = atr(p_bbcode);
```
Same in other places.

Animtim replied:
Thanks, I was wondering the difference between tr() and atr() when I saw it used in other places, now I get it :)

KoBeWi commented:
Need to squash this into 1 commit.

Animtim commented:
Squash (and rebase) done.

timothyqiu commented on scene/gui/rich_text_label.cpp:
```diff
@@ -5114,13 +5114,17 @@ void RichTextLabel::set_text(const String &p_bbcode) {
+	_apply_translated_bbcode(text);
+}
+
+void RichTextLabel::_apply_translated_bbcode(const String &p_bbcode) {
+	String xl_text = atr(p_bbcode);
```
The input is not necessarily BBCode, but is always `text`.

```suggestion
void RichTextLabel::_apply_translation() {
	String xl_text = atr(text);
```

Animtim replied:
Good point, much simpler this way, thanks. Commit updated.

timothyqiu reviewed (approved):
Looks good to me. Left a nitpick :P

akien-mga commented:
Thanks!

Animtim commented:
Yay! :tada: :fireworks:
